pkg/tfbridge/provider.go

```diff
@@ -74,6 +74,7 @@ type Provider struct {
 	pulumiSchema     []byte                             // the JSON-encoded Pulumi schema.
 	pulumiSchemaSpec *pschema.PackageSpec
 	memStats         memStatCollector
+	hasTypeErrors    map[resource.URN]bool
```

Comment:
Can we change this into a set instead of a map? It's marginally more efficient, but more importantly it reduces the state space of `hasTypeErrors`. You don't need to worry what this means:

```go
if has, ok := p.hasTypeErrors[key]; ok && !has {
    panic("What does this mean?")
}
```

```suggestion
	hasTypeErrors    map[resource.URN]struct{}
```